tests/component.test.tsx

```diff
@@ -0,0 +1,89 @@
+import { afterEach, expect, test } from 'bun:test';
+import { cleanup, render, screen } from '@testing-library/react';
+import { userEvent } from '@testing-library/user-event';
+import { create } from 'zustand';
+import { createZustandContext } from '../src';
+import { allCats, newCat } from './mocks';
+
+type Cat = string;
+
+type CatStore = {
+	cats: Cat[];
+	addCat: (cat: Cat) => void;
+	removeCat: (cat: Cat) => void;
+};
+
+const [CatsProvider, useCatsStore] = createZustandContext((initialState: { cats: Cat[] }) =>
+	create<CatStore>((set) => ({
+		cats: initialState.cats,
+		addCat: (cat) => set((state) => ({ cats: [...state.cats, cat] })),
+		removeCat: (cat) => set((state) => ({ cats: state.cats.filter((c) => c !== cat) })),
+	})),
+);
+
+const CatsList = () => {
+	const cats = useCatsStore((state) => state.cats);
+	const addCat = useCatsStore((state) => state.addCat);
+	const removeCat = useCatsStore((state) => state.removeCat);
+
+	return (
+		<div>
+			<ul>
+				{cats.map((cat) => (
+					<li key={cat}>{cat}</li>
+				))}
+			</ul>
+			<button type="button" onClick={() => addCat(newCat)}>
+				Add new cat
+			</button>
+			<button type="button" onClick={() => removeCat(newCat)}>
+				Remove new cat
+			</button>
+		</div>
+	);
+};
+
+const renderApp = () =>
+	render(
+		<CatsProvider initialValue={{ cats: allCats }}>
+			<CatsList />
+		</CatsProvider>,
+	);
+
+afterEach(cleanup);
+
+test('should initialize state correctly', () => {
+	renderApp();
+
+	for (const cat of allCats) {
+		expect(screen.getByText(cat)).toBeTruthy();
+	}
+});
+
+test('should add new cat', async () => {
+	const user = userEvent.setup();
+
+	renderApp();
+
+	await user.click(screen.getByRole('button', { name: 'Add new cat' }));
```

Comment:
seeing as we’re only calling one event per test, we can just use `userEvent.click` instead of setting it up as `user`. 

Are there any advantages to setting up userEvent first?

```suggestion
	await userEvent.click(screen.getByRole('button', { name: 'Add new cat' }));
```

Reply:
Thank you for your feedback, I used it this way because it is recommended in the docs: 
https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent

You can have a look at the last paragraph of this section, in which it is recommended to use the methods on the instances returned by `userEvent.setup()`, instead of directly invoking APIs such as `userEvent.click()`.

Reply:
good to know it's a testing library v14 change that's just supported as they migrate away from it. I've just learnt a new thing and now wanna adjust all my unit tests at work 🤣